extensions/amp-animation/0.1/web-animations.js

```diff
@@ -1179,13 +1191,32 @@ class CssContextImpl {
 
   /** @override */
   getViewportSize() {
-    if (!this.viewportSize_) {
-      this.viewportSize_ = {
-        width: this.win_./*OK*/ innerWidth,
-        height: this.win_./*OK*/ innerHeight,
-      };
+    return this.getViewportParams_().size;
+  }
+
+  /** @private */
+  getViewportParams_() {
+    if (!this.viewportParams_) {
+      if (this.scope_ && this.scaleByScope_) {
+        const rect = this.scope_./*OK*/ getBoundingClientRect();
```

Comment:
Is it worth checking if `this.scope_` is an AMP element, to get the cached measurements through `this.scope_.getLayoutBox()` instead?

Reply:
Filing this as a later optimization, I'm not sure about the nuances yet. (In the current state this has better performance than the current presets btw).